header/header.go

```diff
@@ -227,56 +226,19 @@ func (eh *ExtendedHeader) UnmarshalBinary(data []byte) error {
 	return nil
 }
 
-// MarshalJSON marshals an ExtendedHeader to JSON. The ValidatorSet is wrapped with amino encoding,
-// to be able to unmarshal the crypto.PubKey type back from JSON.
+// MarshalJSON marshals an ExtendedHeader to JSON.
+// Uses tendermint encoder for tendermint compatiblity.
 func (eh *ExtendedHeader) MarshalJSON() ([]byte, error) {
-	type Alias ExtendedHeader
-	validatorSet, err := tmjson.Marshal(eh.ValidatorSet)
-	if err != nil {
-		return nil, err
-	}
-	rawHeader, err := tmjson.Marshal(eh.RawHeader)
-	if err != nil {
-		return nil, err
-	}
-	return json.Marshal(&struct {
-		RawHeader    json.RawMessage `json:"header"`
-		ValidatorSet json.RawMessage `json:"validator_set"`
-		*Alias
-	}{
-		ValidatorSet: validatorSet,
-		RawHeader:    rawHeader,
-		Alias:        (*Alias)(eh),
-	})
+	return tmjson.Marshal(*eh)
```

Comment:
It shouldn't be necessary to pass the value unless there is some quirk in the `tmjson`. Have you tried without `*`?

Reply:
it looks like it was the same case with recursive call as in unmarshal. Just dereferencing hidden that. Switched to alias to be clear what issue is solved there

Reply:
I see